### PR TITLE
refactor: Sync Scoping tree with Forms data

### PR DESCRIPTION
**Pull request by simcha90**
### SUMMARY
This PR add some functionality to Native Filters Modal config:

1. Sync scoping tree in Native filters config modal:
 - Save selected scopes to redux
 - Get selected scopes from redux and populate them to the tree
 - Correct processing scoping per selected filter on left side of modal
 - Fixed Filters Scope in Native Filters form
 - Done some general refactoring in Scope tree

☝️ Scoping tree clone logic of old scope tree that worked with `FilterBox` filters. This tree save data in redux `nativeFilters` domain and in future this data will be used to filter specific charts

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Here example of redux/ui equality after tree items selected:
image (7)
image (6)

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by graceguo-supercat** [comment]
Can you summarize what are changes made in this PR? filter scope components are shared by filter_box and native filter component, we need to understand why these changes are necessary. thanks!

File: superset-frontend/src/dashboard/reducers/types.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+export enum Scoping {
+  all,
+  specific,
+}
+
+/** Chart state of redux */
+export type Chart = {
+  id: number;
+  slice_id: 2107;
```
Comment (graceguo-supercat) [inline superset-frontend/src/dashboard/reducers/types.ts:30]: what is `2107` used for?

Comment (simchaNielsen) [reply superset-frontend/src/dashboard/reducers/types.ts:30]: yes, you are right no need it, updating

**Comment by simchaNielsen** [comment]
> Can you summarize what are changes made in this PR? filter scope components are shared by filter_box and native filter component, we need to understand why these changes are necessary. thanks!

We build this Scoping tree component instead old one, it placed in ConfigModal for native filters, it duplicates behaviour of old Scoping tree, but save it in `nativeFilters` redux location...
It's only part of work, we still need connect dashboard charts to this redux location instead old one
cc @amitmiran137 @villebro

**Review by villebro — APPROVED** [review]
Code looks good and tested to correctly populate redux, thanks for this improvement! In the future let's split up refactoring and functional PRs.
